tests/apollo/test_skvbc.py

```diff
@@ -78,6 +78,44 @@ async def test_state_transfer(self, bft_network,exchange_keys=True):
         await bft_network.force_quorum_including_replica(stale_node)
         await skvbc.assert_successful_put_get(self)
 
+    @with_trio
+    @with_bft_network(start_replica_cmd)
+    async def test_request_missing_data_from_previous_window(self, bft_network, exchange_keys=True):
+        """
+        Test that a replica succeeds to ask for missing info from the former window
+
+        1. Start all nodes and process 149 requests.
+        2. Stop one node and process 2 more requests (this will cause to the remaining
+            replicas to proceeds beyond the checkpoint)
+        3. The node should catchup without executing state transfer.
+        """
+
+        if exchange_keys:
+            await bft_network.do_key_exchange()
+        bft_network.start_all_replicas()
+
+        skvbc = kvbc.SimpleKVBCProtocol(bft_network)
+
+        stale_node = random.choice(
+            bft_network.all_replicas(without={0}))
+        for i in range(151):
+            await skvbc.write_known_kv()
+            if i == 149:
+                bft_network.stop_replica(stale_node)
+
```

Comment:
I think it would be good to verify all up nodes reached to the stable Sequence ID before starting the catching up one, to be sure they shifted their Working Window.